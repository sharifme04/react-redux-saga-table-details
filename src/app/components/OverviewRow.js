import React from "react";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { templateDeleteRequested } from "../actions/templates";

const OverviewRow = ({ template }) => {
  const dispatch = useDispatch();
  const { templateId, name, version, validFrom, validTo } = template;
  return (
    <tr>
      <td>{templateId}</td>
      <td>{name}</td>
      <td>{version}</td>
      <td>{validFrom}</td>
      <td>{validTo}</td>
      <td>
        <Link to={`${templateId}`}>
          <button type="button" className="btn btn-info">
            Details
          </button>
        </Link>
      </td>
      <td>
        <button
          type="button"
          className="btn btn-danger"
          onClick={() => dispatch(templateDeleteRequested(templateId))}
        >
          Delete
        </button>
      </td>
    </tr>
  );
};

OverviewRow.propTypes = {
  template: PropTypes.shape({
    name: PropTypes.string,
    itemplateId: PropTypes.string,
    version: PropTypes.number,
    validFrom: PropTypes.string,
    validTo: PropTypes.string,
  }),
};

export default OverviewRow;
