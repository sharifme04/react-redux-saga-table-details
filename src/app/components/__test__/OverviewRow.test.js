import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import OverviewRow from "../OverviewRow";

const mockStore = configureMockStore();
const store = mockStore({});

const templates = {
  conditions: null,
  name: "basic_user",
  optionalDocuments: {
    assetType: {
      conditions: { topLevelDomain: ".edu" },
      values: { emailAddress: true },
    },
  },
  partnerId: "59c25c00-78cd-11ea-ac9b-bf680aaa6d4b",
  requiredDocuments: {
    assetType: {
      conditions: { topLevelDomain: ".edu" },
      values: { emailAddress: true },
    },
  },
  templateId: "487c5b70-9516-11ea-bbfa-ed3b490e43f0",
  type: "basic",
  validFrom: "04-05-2020",
  validTo: "21-05-2020",
  version: 2,
};

describe("<OverviewRow />", () => {
  test("renders initial", () => {
    const component = shallow(
      <Provider store={store}>
        <OverviewRow templates={templates} />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test("return the default array when there is no data to map through", () => {
    const component = shallow(
      <Provider store={store}>
        <OverviewRow />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test("does not break without value", () => {
    const component = shallow(
      <Provider store={store}>
        <OverviewRow />
      </Provider>
    );
    expect(component.find("Link")).toHaveLength(0);
    expect(component.find("button")).toHaveLength(0);
  });

  test("does not break with an empty array", () => {
    const component = shallow(
      <Provider store={store}>
        <OverviewRow fakeTemplates={[]} />
      </Provider>
    );
    expect(component.find("Link")).toHaveLength(0);
    expect(component.find("button")).toHaveLength(0);
  });
});
