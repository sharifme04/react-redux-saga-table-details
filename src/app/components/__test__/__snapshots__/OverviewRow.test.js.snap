// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverviewRow /> renders initial 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <OverviewRow
    templates={
      Object {
        "conditions": null,
        "name": "basic_user",
        "optionalDocuments": Object {
          "assetType": Object {
            "conditions": Object {
              "topLevelDomain": ".edu",
            },
            "values": Object {
              "emailAddress": true,
            },
          },
        },
        "partnerId": "59c25c00-78cd-11ea-ac9b-bf680aaa6d4b",
        "requiredDocuments": Object {
          "assetType": Object {
            "conditions": Object {
              "topLevelDomain": ".edu",
            },
            "values": Object {
              "emailAddress": true,
            },
          },
        },
        "templateId": "487c5b70-9516-11ea-bbfa-ed3b490e43f0",
        "type": "basic",
        "validFrom": "04-05-2020",
        "validTo": "21-05-2020",
        "version": 2,
      }
    }
  />
</ContextProvider>
`;

exports[`<OverviewRow /> return the default array when there is no data to map through 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <OverviewRow />
</ContextProvider>
`;
